app/src/main/java/protect/card_locker/importexport/CatimaImporter.java

```diff
@@ -362,22 +487,50 @@ private void importLoyaltyCard(Context context, SQLiteDatabase database, CSVReco
      * Import a single group into the database using the given
      * session.
      */
-    private void importGroup(SQLiteDatabase database, CSVRecord record) throws FormatException {
-        String id = CSVHelpers.extractString(DBHelper.LoyaltyCardDbGroups.ID, record, null);
+    private void importGroupV2(SQLiteDatabase database, CSVRecord record) throws FormatException {
+        String name = CSVHelpers.extractString(DBHelper.LoyaltyCardDbGroups.ID, record, null);
+        DBHelper.insertGroup(database, name);
+    }
 
-        DBHelper.insertGroup(database, id);
+    /**
+     * Import a single group from the V3 scheme (database v16) into the database using the given
+     * session.
+     */
+    private Pair<Integer, String> importGroupV3(SQLiteDatabase database, CSVRecord record) throws FormatException {
+        Integer id = CSVHelpers.extractInt(DBHelper.LoyaltyCardDbGroups.ID, record, false);
+        String name = CSVHelpers.extractString(DBHelper.LoyaltyCardDbGroups.NAME, record, null);
+        DBHelper.insertGroup(database, name);
```

Comment:
While I may have messed up before and forgotten to import the order once order got added, we should really fix that now and also import the order.

Reply:
It should be solved now, but my implementation should be checked again.